tests/Console/ConfigurationResolverTest.php

```diff
@@ -246,11 +246,53 @@ public function testResolveConfigFileChooseFileWithInvalidFile()
     public function testResolvePathRelative()
     {
         $this->resolver
-            ->setCwd(__DIR__)
-            ->setOption('path', 'Foo'.DIRECTORY_SEPARATOR.'Bar')
+            ->setCwd(dirname(__DIR__))
+            ->setOption('path', basename(__DIR__))
             ->resolve();
 
-        $this->assertSame(__DIR__.DIRECTORY_SEPARATOR.'Foo'.DIRECTORY_SEPARATOR.'Bar', $this->resolver->getPath());
+        $this->assertSame(__DIR__, $this->resolver->getPath());
+    }
+
+    public function testResolvePathWithFileThatIsExcludedDirectly()
+    {
+        $dir = realpath(__DIR__);
+        $this->config->getFinder()
+            ->in($dir)
+            ->notPath(substr(__FILE__, strlen($dir) + 1));
+
+        $this->resolver
+            ->setOption('path', __FILE__)
+            ->resolve();
+
+        $this->assertCount(0, iterator_to_array($this->resolver->getConfig()->getFinder()));
+    }
+
+    public function testResolvePathWithFileThatIsExcludedByDir()
+    {
+        $dir = realpath(__DIR__.'/..');
+        $this->config->getFinder()
+            ->in($dir)
+            ->exclude(substr(realpath(__DIR__), strlen($dir) + 1));
```

Comment:
hmm, symlinks should be resolved in `__DIR__`, so it should be always equivalent to `realpath(__DIR__)`.

http://php.net/manual/en/language.constants.predefined.php

`__FILE__`:
> The full path and filename of the file **with symlinks resolved**.

`__DIR__`:
> This is equivalent to `dirname(__FILE__)`.